### PR TITLE
feat(das): add metric for task queue depth

### PR DESCRIPTION
**Pull request by tahsintunan**

### CODE REVIEW

File: nft_ingester/src/tasks/mod.rs [hunk 1]
```diff
@@ -191,6 +191,26 @@ impl TaskManager {
             .map_err(|e| e.into())
     }
 
+    pub async fn get_task_queue_depth(conn: &DatabaseConnection) -> Result<u64, IngesterError> {
+        tasks::Entity::find()
+            .filter(
+                Condition::all()
+                    .add(tasks::Column::Status.ne(TaskStatus::Success))
```
Comment (nikhil-helius) [inline nft_ingester/src/tasks/mod.rs:198]:
```suggestion
                    .add(tasks::Column::Status.eq(TaskStatus::Pending))
```
not equal to success will also count `failed` tasks which doesn't account for the queue depth

File: nft_ingester/src/tasks/mod.rs [hunk 2]
```diff
@@ -191,6 +191,26 @@ impl TaskManager {
...
+                    .add(
+                        Condition::any()
+                            .add(tasks::Column::LockedUntil.lte(Utc::now()))
+                            .add(tasks::Column::LockedUntil.is_null()),
+                    )
+                    .add(
+                        Expr::col(tasks::Column::Attempts)
+                            .less_than(Expr::col(tasks::Column::MaxAttempts)),
```
Comment (nikhil-helius) [inline nft_ingester/src/tasks/mod.rs:206]: No need to have this condition if you're checking for `pending`.

File: nft_ingester/src/tasks/mod.rs [hunk 3]
```diff
@@ -191,6 +191,26 @@ impl TaskManager {
...
     }
 
+    pub async fn get_task_queue_depth(conn: &DatabaseConnection) -> Result<u64, IngesterError> {
+        tasks::Entity::find()
+            .filter(
+                Condition::all()
+                    .add(tasks::Column::Status.ne(TaskStatus::Success))
+                    .add(
+                        Condition::any()
+                            .add(tasks::Column::LockedUntil.lte(Utc::now()))
```
Comment (nikhil-helius) [inline nft_ingester/src/tasks/mod.rs:201]: Don't think this check is required.

File: nft_ingester/src/tasks/mod.rs (hunk 1 above)
Comment (tahsintunan) [reply nft_ingester/src/tasks/mod.rs:198]: Good catch!

Comment (tahsintunan) [reply nft_ingester/src/tasks/mod.rs:198]: What about `Running` tasks? Don't they also contribute to queue depth?

**Review by nikhil-helius — APPROVED** [review]

**Pull request merged**